### DISCUSSION TITLE
Setting up versatile thermostat in existing configuration

### DISCUSSION THREAD

**Discussion by FritsOV** [jmcollin78/versatile_thermostat#231] [Q&A]
Hello I have a two-storey building and use a heat pump with buffer tank for heating. Heating is done on floor 1 by underfloor heating and 3 fan coils. The heat pump heats a buffer supply of water with regards to outside temperature and desired inside temperature. One circulation pump regulates the hot water flow tot he underfloor and fan coils. The thermostats in this room determine whether the circulation pump should run or not. So if there is a demand for heat in a room, the valves in that room will open and start the circulation pump. If there is no heat demand, the automation will check whether 1 of the thermostats has a heat demand and the circulation pump will remain on, otherwise it will be switched off. This is done by an automation wich looks to the state of the hvac_action of the climate entity: idle, heating or cooling. This works well for over two years now.

On floor two I only have fan coils for heating. Thermostats are the beco bac-006EMLW wich opens valves and has a 0-10v modulating for the fan. The thermostat type doesn't give me any attributes of change of status idle heating or cooling. As a result, I can't have the existing automation check whether there is a heat demand or not. As a result, I cannot turn the circulation pump on or off for this floor. I'm looking for advice on how best to configure or set up floor two with the help of a versatile thermostat in combination with the switch fort he circulation pump. 
My first geuss is to let versatile regulate the themperature by the switch of the thermostat but also with a second switch for the circulation pump. I guess i also have to get the state change from the versatile thermostat in combination with floor 1 and the automation as i described. 

Can i get some advice how best to proceed?

**Comment by FritsOV** [comment]
I created versatile over thermostats. 
as shown in the picture the status of the thermostat is heating. But that is not true. 
Where is the attribute for the change state from heating to idle, is there any?
i need this to shut down the pump or to start it up. please any help would be apreciated. 

![thermostat](https://github.com/jmcollin78/versatile_thermostat/assets/76515350/4eddc16f-0ecb-40d3-a977-b99b673c206b)

![attributes](https://github.com/jmcollin78/versatile_thermostat/assets/76515350/0e4c25d2-46ae-4c64-8e68-62a42f73d3bb)

**Reply by jmcollin78** [reply]
Hello, @FritsOV,

In over_climate, the hvac_action attributes comes from the underlying climate (climate.kantoor in your example). VTherm is not able to know if the underlying is really heating or not so it is relying on the underlying info.

**Comment by FritsOV** [comment]
Thank you for replying. 

Would it be possible to just get the temperature from the thermostat and use the function over switch.,
Defining an switch helper in HA  that turns on the circulation pump in stead of a heater?

**Reply by jmcollin78** [reply]
I think not. The over_switch aims to regulate the power by switching on and off continuously the switch until the target temp is reach.
or I don't get you maybe

**Comment by FritsOV** [comment]
there is one circulation pump. The pump will only start if there is a demand of heat in the rooms where thermostats are placed. 
There are 6 thermostats. The problem with two is that they dont give a atrribute that indicates that the thermostat is idle or in heat. These two are always in heat (or cool). 
i can see if i use the thermostat over switch that there is an attribute idle or heat. Over switch i can therefore turn on and or off the circulation pump. And then check in an automation if another thermostat is in heat or in idle state to defenitly shut down the pump or leave it on. 

The problem i am now facing is that i need the combination of the two. because over switch does not regulate the thermostat. And i cant bind versatile over climate to versatile over switch.

**Reply by jmcollin78** [reply]
For me it is not possible to control your circulation pump with a VTherm over_switch. Over switch are working like that :
![Capture d’écran 2023-11-29 à 08 47 33](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/1bb8e1d1-d96a-425f-a0d0-2eeba89cfd91)

I'm pretty sure you don't you to start / stop every 5 minutes your circulation pump.

**Reply by FritsOV** [reply]
I dont get the impression that it is working like you sugest. 
Let me explain what i have so far.
1. climate entity with tuya local  -> climate.kantoor
2. climate over switch with versatile -> climate.office_switch
3. dummy switch for climate over switch climate.office_switch

automation:
if state climate.office_switch changes from idle to heat:

1.   set circulation pump  to on, 
2.  call service HVAC  set climate.kantoor to heat   (thermostat is turned on)
3.  call service HVAC  set climate.kantoor to same setting as climate.office_switch e.g. eco, comfort, boost. 

if state climate. office_switch changes from heat to idle:

1. set circulation pump dummy switch to off 
-  call automation to set switch.CV_pump to off
-  check if one of the other thermostats are idle or in heat mode.
             -  if NOT in heat mode  set switch. CV_pump to off
             -  Call service HVAC en set HVAC  climate.kantoor  to off -> thermostat is off
             
             - If any thermostat is in heat mode leave Switch.CV_pump to its state (ON)
             -  call service HVAC en set climate.kantoor to off

However this works for now, changes for the versatile over switch should schange the thermostat in the same way. 
I was wondering what your thoughts are for this sollution?

Kind regards

**Reply by jmcollin78** [reply]
ok that was not clear to me until this post.

Managing the pump with the hvac_action status of the VTherms is certainmy the best solution. I will add a § in the readme for this.

See and comment this if you want: https://github.com/jmcollin78/versatile_thermostat/issues/234

**Comment by FritsOV** [comment]
Okay thank you for your answer.
My last question is: The thermostat over switch has a step mode of 0.1 for changing the temperature  how can i change this in e.g. 0.5

**Reply by jmcollin78** [reply]
In over switch, yes the step mode is 0.1 and not configurable now. This is the default step mode in HA.

**Comment by FritsOV** [comment]
Thanks. would be nice if it was configurable. I also noticed a BUG. 
Setting AC mod in versatile over switch and defining the defaults for eco, comfort en boost for heat and cool. 
If device is in heat HVACmode it gives for modes: boost, comfort and eco the cool predefined setting,  it does not do anything with predefined settings in heat HVAC mode.

**Reply by jmcollin78** [reply]
I have already seen an issue on this and it is fixed in the next release.
